content/en/blog/2021/wasm-api-alpha/index.md

```diff
@@ -0,0 +1,121 @@
+  0:  0xdbd - runtime._panic
+  1:  0x103ab - main.anotherCalculation
+  2:  0x10399 - main.someCalculation
+  3:  0xea57 - main.myHeaderHandler
+  4:  0xea15 - proxy_on_request_headers
+{{< /text >}}
+
+The above is an example output of the stack trace for Go SDK based Wasm extensions. You might notice that you want file names and lines in the trace lines -- that is another huge future work and is currently an open issue since it has something to do with the DWARF format for WebAssembly and the Exception Handling proposal to WebAssembly specification.
+
+### Strace support for Wasm programs
+
+You can see strace equivalent logs emitted by Envoy. With Istio proxy’s component log level `wasm:trace`, we can observe all the system calls and Proxy-Wasm ABI calls that go across the boundary between Wasm virtual machines and Envoy. The following is an example log stream of such strace logs:
+
+{{< text plain >}}
+[host->vm] proxy_on_context_create(2, 1)
+[host<-vm] proxy_on_context_create return: void
+[host->vm] proxy_on_request_headers(2, 8, 1)
+[vm->host] wasi_snapshot_preview1.random_get(86928, 32)
+[vm<-host] wasi_snapshot_preview1.random_get return: 0
+[vm->host] env.proxy_log(2, 87776, 18)
+{{< /text >}}
+
+This is especially useful to debug Wasm program’s execution at runtime, for example, to verify it is not making any malicious system calls.
+
+### Arbitrary Prometheus namespace for in-Wasm metrics
+
+The next one is about metrics. Wasm extensions have been able to define their own custom metrics and expose them in Envoy just like any other metric, but prior to Istio 1.12, all of these custom metrics are prefixed by `envoy_` Prometheus namespace and users were not able to have their own namespaces. Now, you can choose whatever namespace you want, and your metrics are exposed in Envoy as-is without being prefixed by `envoy_`.
+
+Note that in order to actually expose these custom metrics, you have to configure [`ProxyConfig.proxyStatsMatcher`](../../../docs/reference/config/istio.mesh.v1alpha1/#ProxyConfig-ProxyStatsMatcher) in `meshConfig` for global configuration or in `proxy.istio.io/config` for per proxy configuration. For detail, please refer to [`Envoy Statistics`](../../../docs/ops/configuration/telemetry/envoy-stats/).
+
+## Future work and looking for feedback
+
+Although we have announced the alpha availability of Wasm plugins, there are still a lot of aspects that are left to be completed. The one important thing is the "Image pull secrets” support in the Wasm API -- this way we would be able to consume the OCI images in the private repository easily in Istio. Others include the first-class support for L4 filters, signature verification of Wasm binaries, runtime improvements in Envoy, Proxy-Wasm SDK improvements, documentation, etc.
+
+That means, this is just the beginning of the 1st-class Wasm support in Istio. We would love to hear feedback from users so that we could improve the developer experience with the Wasm plugins!
```

Comment:
```suggestion
This is just the beginning of our plan to provide 1st-class Wasm support in Istio. We would love to hear your feedback so that we can improve the developer experience using Wasm plugins, in future releases of Istio!
```